### PR TITLE
fix(runtime): do not panic on irregular dir entries

### LINKED ISSUE #9578: Deno.readDir("/dev/fd") is panic
```
❯ deno --version
deno 1.7.5 (release, x86_64-apple-darwin)
v8 9.0.123
typescript 4.1.4

❯ deno eval 'Deno.readDir("/dev/fd")'
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 9, kind: Other, message: "Bad file descriptor" }', runtime/ops/fs.rs:1059:43
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: JoinError::Panic(...)', runtime/ops/fs.rs:1077:4
```

### PR DESCRIPTION
**Pull request by caspervonb**
Fixes #9578

### CODE REVIEW

File: cli/dts/lib.deno.ns.d.ts
Comment (bartlomieju) [inline cli/dts/lib.deno.ns.d.ts:1423]: This is a breaking change

Comment (caspervonb) [reply cli/dts/lib.deno.ns.d.ts:1423]: Right; initially I was going to go with just having everything be false here and on second thought I think that makes alot more sense.

Deno.stat returns all as false without panicking.

```js
> Deno.statSync("/dev/fd/2")
{
  isFile: false,
  isDirectory: false,
  isSymlink: false,
  size: 0,
  mtime: 2021-02-22T14:55:17.914Z,
  atime: 2021-02-22T14:55:17.914Z,
  birthtime: 1970-01-01T00:00:00.000Z,
  dev: 608056912,
  ino: 649,
  mode: 8592,
  nlink: 1,
  uid: 501,
  gid: 4,
  rdev: 268435457,
  blksize: 131072,
  blocks: 0
}
```

Comment (caspervonb) [reply cli/dts/lib.deno.ns.d.ts]: Went back to my first pass on this which is just returning false for all fields.

**Review by bartlomieju — APPROVED** [review]
I'm in favor of this change, waiting on second opinion before landing

**Review by lucacasonato — APPROVED** [review]
LGTM. I am also in favor of this change.

**Review by ry — APPROVED** [review]
LGTM - thanks @caspervonb

**Pull request merged**
